### PR TITLE
[FIX] Feature constructor Python 3.5 ast support

### PR DESCRIPTION
**Pull request by ales-erjavec**
* Fix an error in Feature Constructor widget due to a change in Python 3.5 ast (`AttributeError: 'Call' object has no attribute 'starargs'`)
* Add tests
* Some other minor fixes and code cleanup

### CODE REVIEW

File: Orange/widgets/data/owfeatureconstructor.py [hunk 1]
```diff
@@ -541,8 +524,14 @@ def apply(self):
 
         desc = list(self.featuremodel)
 
+        def validate(source):
+            try:
+                return validate_exp(ast.parse(source, mode="eval"))
```
Comment (kernc) [inline Orange/widgets/data/owfeatureconstructor.py]: Could't we just eval the expression and see if it fails with SyntaxError while the rest is on the user?

File: Orange/widgets/data/owfeatureconstructor.py [hunk 2]
```diff
@@ -663,6 +652,68 @@ def freevars(exp, env):
         raise ValueError(exp)
 
 
+def validate_exp(exp):
+    """
+    Validate an `ast.AST` expression.
+
+    Only expressions with no list,set,dict,generator comprehensions
```
Comment (kernc) [inline Orange/widgets/data/owfeatureconstructor.py]: I've been meaning to introduce a ListVariable type. Seems like I wouldn't be able to populate it with feature constructor?

Comment (kernc) [inline Orange/widgets/data/owfeatureconstructor.py]: I see other uses cases where the expression evaluated to a list of values would be useful (like setting the values of all instances at once (though for that I'd also need to be given all instances at once)). How might constructing a `np.diff` work?

Comment (ales-erjavec) [inline Orange/widgets/data/owfeatureconstructor.py]: > I've been meaning to introduce a ListVariable type. Seems like I wouldn't be able to populate it with feature constructor?

No, unless you enable/change the accepted syntax for your variable type.

> I see other uses cases where the expression evaluated to a list of values would be useful (like setting the values of all instances at once (though for that I'd also need to be given all instances at once))

That should be done by a vectorized expression, which is an argument for limiting the syntax further.

**Pull request merged**